src/auth/googleauth.ts

```diff
@@ -123,182 +151,182 @@ export class GoogleAuth {
     // implemented yet)
 
     if (this._cachedProjectId) {
-      if (callback) setImmediate(callback, null, this._cachedProjectId);
-    } else {
-      const myCallback = (err?: Error|null, projectId?: string) => {
-        if (!err && projectId) {
-          this._cachedProjectId = projectId;
-        }
-        if (callback) setImmediate(callback, err, projectId);
-      };
+      return this._cachedProjectId;
+    }
 
-      // environment variable
-      if (this._getProductionProjectId(myCallback)) {
-        return;
-      }
+    let projectId: string|null|undefined;
 
-      // json file
-      this.getFileProjectId((err?: Error|null, projectId?: string) => {
-        if (err || projectId) {
-          myCallback(err, projectId);
-          return;
-        }
+    // environment variable
+    projectId = this.getProductionProjectId();
```

Comment:
This code here will be simplified a lot if you used the following pattern:

```ts
projectId = this.getProductionProjectId() || await this.getFileProjectId() | await this.getDefaultServiceProjectId() || ...
if (projectId)
```

Reply:
not sure what I was thinking there lol